docs/cpu_templates/cpuid-normalization.md

```diff
@@ -0,0 +1,80 @@
+# CPUID normalization (x86_64 only)
+
+On x86_64, Firecracker makes certain modifications to the guest's CPUID
+regardless of whether a CPU template is used. This is referred to as
+`CPUID normalization`. If a CPU template is used the normalization is
+performed _after_ the CPU template is applied. That means that if the CPU
+template configures CPUID bits used in the normalization process, they will
+be overwritten.
+
+See also: [boot protocol settings](boot-protocol.md)
+
+## x86_64 common CPUID normalization
```

Comment:
nit: Could we use a table for these modifications?

e.g.


description|leaf|register|bit/s
---|---|---|---
Set CLFLUSH line size|0x1|EBX|15:8
Set maximum number of addressable IDs for logical processors in the physical package|0x1|EBX|25:16
Set initial APIC ID|0x1|EBX|31:24
Enable TSC_DEADLINE|0x1|ECX|24
Enable HYPERVISOR|0x1|ECX|31
Set HTT value if the microVM's CPU count is greater than 1|0x1|EDX|28
Insert leaf 0xb, subleaf 0x1 filled with `0` if it is not already present|0xb||
Update extended topology enumeration in all subleaves of leaf 0xb (EAX[4:0], EBX[15:0], ECX[15:0])|0xb||

Reply:
good idea, done